### ISSUE TITLE
Invalid UTF-8 string after upgrade macOS to 14.4

### ISSUE THREAD

**Issue by patanachai** [minetest/minetest#14463]
### Minetest version

```true
Minetest 5.8.0 (OSX)
Using Irrlicht 1.9.0mt13
Using LuaJIT 2.1.1700008891
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_CURL=1
USE_GETTEXT=1
USE_SOUND=1
STATIC_SHAREDIR="minetest.app/Contents/Resources"
STATIC_LOCALEDIR="minetest.app/Contents/Resources/locale"
```

### Operating system and version

macOS Sonoma 14.4

### CPU model

Apple M1 Pro

### Summary

The issue happens after upgrade macOS from 14.3 to 14.4.
I tried the zip file installation and `brew install minetest`
![430773964_424515106903221_7139499169734611224_n](https://github.com/minetest/minetest/assets/963811/ad5fd3b1-f407-4328-a36c-06d8ad8f7f7f)

### Steps to reproduce

- Upgrade macOS to version 14.4
- install minetest
- launch minetest

**Comment by SmallJoker** [comment]
Relevant code:

https://github.com/minetest/minetest/blob/d5d6e36ae0790c5e6b6e896ee04726ae5d6cda09/src/util/string.cpp#L73-L135

I find it strange that a macOS upgrade would change the behaviour of iconv (for encoding conversion). I'd assume `DEFAULT_ENCODING = "WCHAR_T"` is no longer accepted. Does the test `TestUtilities::testUTF8` in `minetest --run-unittests` fail too?

**Comment by sfan5** [comment]
Can you test the artifacts from [here](https://github.com/minetest/minetest/actions/runs/8294735036) and [here](https://github.com/minetest/minetest/actions/runs/8308623588?pr=14464) and tell us if either works?

**Comment by sfan5** [comment]
(reply here -> https://github.com/minetest/minetest/pull/14464#issuecomment-2002088792)

**Comment by prg3** [comment]
PR from 14464 worked for me to fix this problem

**Comment by TechDudie** [comment]
can confirm issue
updated to 14.4 today
<img width="1680" alt="image" src="https://github.com/minetest/minetest/assets/73961295/9b590d0c-a4a7-4b82-b980-ff1c54923bd9">

**Comment by Zephiii** [comment]
Confirmed on macOS 14.5

**Comment by knownasilya** [comment]
The latest mac download from the site has this issue, am I doing something incorrectly?

**Comment by appgurueu** [comment]
The MacOS releases linked on minetest.net are for version 5.8.0, which has this issue. The PR which fixed this (https://github.com/minetest/minetest/pull/14464) landed in 5.9-dev, which has not yet been released, but will be released soon.

In the meantime, you could try the artifacts produced by a recent CI run: https://github.com/minetest/minetest/actions/runs/10237764478/artifacts/1773884578
